openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -1018,3 +1081,25 @@ def get_containers_with_entity(
     #     publishable_entity__draft__version__containerversion__entity_list__in=lists
     # )
     return qs
+
+
+def get_container_children_count(
+    container: Container,
+    *,
+    published: bool,
+):
+    """
+    [ 🛑 UNSTABLE ]
+    Get the count of entities in the current draft or published version of the given container.
+
+    Args:
+        container: The Container, e.g. returned by `get_container()`
+        published: `True` if we want the published version of the container, or
+            `False` for the draft version.
+    """
+    assert isinstance(container, Container)
+    container_version = container.versioning.published if published else container.versioning.draft
+    if container_version is None:
+        raise ContainerVersion.DoesNotExist  # This container has not been published yet, or has been deleted.
+    assert isinstance(container_version, ContainerVersion)
+    return container_version.entity_list.entitylistrow_set.count()
```

Comment:
Does this work in the edge case where we've soft-deleted a child element and published that soft-delete, but haven't removed that element from the container? Don't we need to filter those out of the entry list?

Reply:
Ah great catch, thank you @ormsbee . We should add a test for this case too.

Reply:
Yes, thank you. Updated: a11f1bdfee83